### DISCUSSION TITLE
load balance of adaptive refinement in libmesh

### DISCUSSION THREAD

**Discussion by AIBSCT80** [libMesh/libmesh#3530] [General]
Hi, 

may I sincerely ask in libmesh,  when using adaptive mesh refinement without hanging nodes, will the parallel take care of its load-balancing?

Thanks

**Comment by roystgnr** [comment]
We don't have anything built-in that does AMR without hanging nodes (or their equivalent with hierarchical p-refinement: edge functions constrained to be 0).

But regardless of whether you use built-in or 3rd party refinement, the load balancing gets done whenever the mesh is repartitioned.  By default that gets done every time a changed mesh is reprepared for use, i.e. after every refinement step, but some user codes do it less frequently, on the theory that the calculation and communication involved is only worth it when the load imbalance becomes sufficiently large.

**Comment by AIBSCT80** [comment]
Thanks a lot for the information.  I was looking for some source to perform adaptive refinement without hanging nodes in parallel. Most likely this is not an easy task. I am not sure whether there is any code doing this.  You have right, load balancing can be always performed after the mesh is refined.  I hope I can practice a bit libmesh for my adaptive fem applications.  Thanks
